test/Microsoft.AspNetCore.Razor.Evolution.Test/CaseSensitiveTagMatchingRuleComparer.cs

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Linq;
+using Microsoft.AspNetCore.Razor.Evolution.Legacy;
+using Microsoft.Extensions.Internal;
+using Xunit;
+
+namespace Microsoft.AspNetCore.Razor.Evolution
+{
+    internal class CaseSensitiveTagMatchingRuleComparer : TagMatchingRuleComparer
```

Comment:
There's value. Having multiple `HtmlTargetElementAttribute`s with different cases may pass on the product variant of `TagMatchingRuleComparer` but not the case sensitive one depending on if we consume the attributes correctly.